twenty-first/src/math/digest.rs

```diff
@@ -11,18 +11,19 @@ use rand::Rng;
 use rand_distr::Distribution;
 use rand_distr::Standard;
 use serde::Deserialize;
+use serde::Deserializer;
 use serde::Serialize;
+use serde::Serializer;
 
-use crate::error::TryFromDigestError;
+use crate::error::{FromHexDigestError, TryFromDigestError};
```

Comment:
yeah I've been drifting towards your position over time.   The primary reason is that the single-value-per-line is much easier to work with when resolving git conflicts.

I think the counterpoints would be that single-value-per-line is more verbose and may obscure (at a glance) that types come from the same crate or module.

That said, for my workflow, the git-conflict (diff) thing outweighs the counterpoints, so I'm happy to make the change and use that style going forward.

At present, we use a mixed style, at least in neptune-core.  It would probably be good to settle on a style across neptune-crates.   I wonder if there is tooling support for this, eg a `cargo fmt` flag, clippy lint, etc.   Ideally, `cargo fmt` or similar tool would just reformat each crate for us...

Reply:
Actually, I concur with the git conflicts being the primary reason to avoid this. Seems like I have forgotten how annoying that is. (I'm using single-line style in the crates I mainly work on for a while now.)

I'd be stoked to learn about a tool that automates this conversion. Currently, I basically run `grep "^ *use.*\{"` to see if any violations have snuck in, which is admittedly a terrible workflow.

Reply:
From what I can tell, [this](https://rust-lang.github.io/rustfmt/?version=v1.6.0&search=#imports_granularity) is the relevant option on `rustfmt`. There is an open tracking issue with recent activity linked there.

Reply:
>  It would probably be good to settle on a style across neptune-crates

I also concur with this. As far as I recall, @Sword-Smith is also in favor of the single-line style – please correct me if I misremember. That said, it's very tedious to manually change everything. :face_exhaling: